src/sdk/clients/createBicoPaymasterClient.test.ts

```diff
@@ -123,4 +131,194 @@ describe.runIf(paymasterTruthy())("bico.paymaster", async () => {
+    })
+
+    // Check that the balance hasn't changed
+    // No gas fees were paid, so the balance should have decreased only by 1n
+    expect(finalBalance).toBe(initialBalance - 1n)
+  })
+
+  test("should use token paymaster to pay for gas fees, use max approval, use sendTransaction", async () => {
+    const paymasterContext = toBiconomyTokenPaymasterContext({
+      feeTokenAddress: baseSepoliaUSDCAddress
+    })
+    const nexusClient = await createNexusClient({
+      signer: account,
+      chain,
+      paymaster: createBicoPaymasterClient({
+        transport: http(paymasterUrl)
+      }),
+      paymasterContext,
+      transport: http(),
+      bundlerTransport: http(bundlerUrl),
+      ...testParams
+    })
+
+    const initialBalance = await publicClient.getBalance({
+      address: nexusAccountAddress
+    })
+
+    const tokenPaymasterUserOp = await nexusClient.prepareTokenPaymasterUserOp({
+      calls: [
+        {
+          to: recipientAddress,
+          value: 1n,
+          data: "0x"
+        }
+      ],
+      feeTokenAddress: baseSepoliaUSDCAddress
+    })
+
+    const hash = await nexusClient.sendTransaction(tokenPaymasterUserOp)
+
+    const receipt = await publicClient.waitForTransactionReceipt({ hash })
+
+    expect(receipt.status).toBe("success")
+
+    // Get final balance
+    const finalBalance = await publicClient.getBalance({
+      address: nexusAccountAddress
+    })
+
+    // Check that the balance hasn't changed
+    // No gas fees were paid, so the balance should have decreased only by 1n
+    expect(finalBalance).toBe(initialBalance - 1n)
+  })
+
+  test("should use token paymaster to pay for gas fees, use custom approval with token paymaster quotes", async () => {
+    const paymasterContext = toBiconomyTokenPaymasterContext({
+      feeTokenAddress: baseSepoliaUSDCAddress
+    })
+    const nexusClient = await createNexusClient({
+      signer: account,
+      chain,
+      paymaster: createBicoPaymasterClient({
+        transport: http(paymasterUrl)
+      }),
+      paymasterContext,
+      transport: http(),
+      bundlerTransport: http(bundlerUrl),
+      ...testParams
+    })
+
+    const usdcBalance = await publicClient.readContract({
+      address: baseSepoliaUSDCAddress,
+      abi: parseAbi([
+        "function balanceOf(address owner) public view returns (uint256 balance)"
+      ]),
+      functionName: "balanceOf",
+      args: [nexusClient.account.address]
+    })
+
+    expect(usdcBalance).toBeGreaterThan(0n)
+
+    const initialBalance = await publicClient.getBalance({
+      address: nexusClient.account.address
+    })
+
+    const tokenList = [baseSepoliaUSDCAddress]
+    const userOp = await nexusClient.prepareUserOperation({
+      calls: [
+        {
+          to: recipientAddress,
+          value: 1n,
+          data: "0x"
+        }
+      ]
+    })
+    const quote = await paymaster.getTokenPaymasterQuotes({ userOp, tokenList })
+    const usdcFeeAmount = parseUnits(
```

Comment:
To me this reads like the quote is not used and that the amount might ultimately be a bit different. It reads like the quote is acquired, and then discarded

Reply:
This could cause sporadic approval failures if this is doing what I think it is doing.